examples/tensorflow/translation/run_translation.py

```diff
@@ -0,0 +1,625 @@
+        num_update_steps_per_epoch = len(train_dataset) // training_args.per_device_train_batch_size
+        num_train_steps = training_args.num_train_epochs * num_update_steps_per_epoch
+        optimizer, lr_schedule = create_optimizer(
+            init_lr=training_args.learning_rate,
+            num_train_steps=num_train_steps,
+            num_warmup_steps=training_args.warmup_steps,
+        )
+
+        def masked_sparse_categorical_crossentropy(y_true, y_pred):
+            # We clip the negative labels to 0 to avoid NaNs appearing in the output and
+            # fouling up everything that comes afterwards. The loss values corresponding to clipped values
+            # will be masked later anyway, but even masked NaNs seem to cause overflows for some reason.
+            # 1e6 is chosen as a reasonable upper bound for the number of token indices - in the unlikely
+            # event that you have more than 1 million tokens in your vocabulary, consider increasing this value.
+            # More pragmatically, consider redesigning your tokenizer.
+            losses = tf.keras.losses.sparse_categorical_crossentropy(
+                tf.clip_by_value(y_true, 0, int(1e6)), y_pred, from_logits=True
+            )
+            # Compute the per-sample loss only over the unmasked tokens
+            losses = tf.ragged.boolean_mask(losses, y_true != -100)
+            losses = tf.reduce_mean(losses, axis=-1)
+            return losses
+
+        # endregion
+
+        # region Metric and postprocessing
+        metric = load_metric("sacrebleu")
+
+        def postprocess_text(preds, labels):
+            preds = [pred.strip() for pred in preds]
+            labels = [[label.strip()] for label in labels]
+
+            return preds, labels
+
+        # endregion
+
+        # region Training
+        model.compile(loss={"logits": masked_sparse_categorical_crossentropy}, optimizer=optimizer)
+
+        if training_args.do_train:
+            logger.info("***** Running training *****")
+            logger.info(f"  Num examples = {len(train_dataset)}")
+            logger.info(f"  Num Epochs = {training_args.num_train_epochs}")
+            logger.info(f"  Instantaneous batch size per device = {training_args.per_device_train_batch_size}")
+            logger.info(f"  Total train batch size = {total_train_batch_size}")
+            logger.info(f"  Total optimization steps = {num_train_steps}")
+
+            model.fit(
+                tf_train_dataset,
+                epochs=int(training_args.num_train_epochs),
+                steps_per_epoch=num_update_steps_per_epoch,
+            )
+        # endregion
+
+        # region Validation
+        if data_args.val_max_target_length is None:
+            data_args.val_max_target_length = data_args.max_target_length
+
+        gen_kwargs = {
+            "max_length": data_args.val_max_target_length if data_args is not None else config.max_length,
```

Comment:
The test is weird, is it `data_args.val_max_target_length`? Also that number has been set just above.

Reply:
A lot of this is just copied from the PT scripts, but this bit does look like it got a bit mangled. I've simplified it!